src/databricks/labs/ucx/installer/workflows.py

```diff
@@ -356,6 +356,9 @@ def _infer_task_exception(haystack: str) -> Exception:
     def _upload_wheel(self):
         with self._wheels:
             try:
+                #TODO: add a flag to upload dependencies to dbfs or wsfs
+                #TODO: the version.json file gets the first wheel in the list of the wheel packages - this should be the main UCX wheel
+                #TODO: the wheel packages installed are compatible with the host machine and not the  target DB clustersgit , example: charset_normalizer-3.3.2-cp310-cp310-macosx_11_0_arm64.whl
```

Comment:
most likely you need to extend this place:

https://github.com/databrickslabs/blueprint/blob/a5a85631b46207ee1cec4b3e3cddae14e2b1aedf/src/databricks/labs/blueprint/wheels.py#L269-L277

Reply:
A draft PR raised to Blueprint: https://github.com/databrickslabs/blueprint/pull/99